src/Security/OAuthUserProvider.php

```diff
@@ -25,20 +27,33 @@
 final class OAuthUserProvider implements OAuthAwareUserProviderInterface
 {
     /**
-     * @var Registry
+     * @var RegistryInterface
      */
     private $doctrine;
 
-    public function __construct(Registry $doctrine)
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    /**
+     * @var UserResponseInterface
+     */
+    private $response;
+
+    public function __construct(RegistryInterface $doctrine, EventDispatcherInterface $eventDispatcher)
     {
-        $this->doctrine = $doctrine;
+        $this->doctrine        = $doctrine;
+        $this->eventDispatcher = $eventDispatcher;
     }
 
     public function loadUserByOAuthUserResponse(UserResponseInterface $response)
     {
-        $name     = $response->getResourceOwner()->getName();
-        $username = $response->getUsername();
-        $email    = $response->getEmail();
+        $this->response = $response;
```

Comment:
Why don't you pass the whole response to loadDatabaseUser, instead of setting this in a property, which makes it tempting to start using `$this->response` on any place in this class.